pkg/tools/cli/loadtest/service/loader.go

```diff
@@ -66,13 +69,21 @@ func NewLoader(opts ...Option) (Loader, error) {
 		}
 	}
 
+	var err error
 	switch l.operation {
 	case config.Insert:
-		l.requestsFunc, l.loaderFunc = newInsert()
+		l.loaderFunc, err = l.newInsert()
+	case config.StreamInsert:
+		l.loaderFunc, err = l.newStreamInsert()
 	case config.Search:
-		l.requestsFunc, l.loaderFunc = newSearch()
+		l.loaderFunc, err = l.newSearch()
+	case config.StreamSearch:
+		l.loaderFunc, err = l.newStreamSearch()
 	default:
-		return nil, errors.Errorf("undefined method: %v", l.operation)
+		err = errors.Errorf("undefined operation: %s", l.operation.String())
+	}
+	if err != nil {
+		return nil, err
```

Comment:
I think applying the comments below will make things simpler.
https://github.com/vdaas/vald/pull/552/files#r462742394

for example.
```
l.loaderFunc, err = provider.LoaderFunc()
if err != nil {
    return nil, err
}
```

Reply:
If this PR includes fix about provider, I think it is too large.

Reply:
ok, It seems to be good in the direction of responding with another PR.

cc: @kpango